gamedata/movedefs.lua

```diff
@@ -1,478 +1,516 @@
 ----------------------------------------------------------------------------------
---README
---for organizational purposes all unit deffs must be added to thge movedeff name.
--- formatted as such
---
-	-- armfav/corfav                   <-- add unitdeffname here for the below movedeff type
-	--TANK1 = {
-	--	crushstrength = 10,
-	--	footprintx = 2,
-	--	footprintz = 2,
-	--	maxslope = 18,
-	--	slopeMod = 18,
-	--	maxwaterdepth = 22,
-	--	depthModParams = {
-	--		minHeight = 4,
-	--		linearCoeff = 0.03,
-	--		maxValue = 0.7,
-	--	}
-	--},
+-- README
+-- For organizational purposes all unit deffs must be added to thge movedeff name.
+-- Formatted as such:
+
+-- -- armfav corfav        <-- add internal names above for the below movedef type
+-- TANK1 = {
+--    crushstrength = 10,
+--    footprintx = 1,      <-- footprint size must match moveDef name
+--    footprintz = 1,      <-- and must be square
+--    maxslope = 18,
+--    slopeMod = 18,
+--    maxwaterdepth = 22,
+--    depthModParams = {
+--        minHeight = 4,
+--        linearCoeff = 0.03,
+--        maxValue = 0.7,
+--    }
+-- },
 
 ----------------------------------------------------------------------------------
+-- REFERENCE VALUES
+-- These are for readability more than correctness, really.
+
+local CRUSH = {
+	NONE    = 0,
+	TINY    = 5,
+	LIGHT   = 10, -- default
+	SMALL   = 18,
+	MEDIUM  = 25,
+	LARGE   = 50,
+	HEAVY   = 250,
+	HUGE    = 1400,
+	MASSIVE = 9999,
+	MAXIMUM = 99999, -- arbitrary limit
+}
+
+local DEPTH = {
+	NONE       = 0,
+	SHALLOW    = 10, -- add unit size tolerances to this value
+	SUBMERGED  = 15,
+	AMPHIBIOUS = 5000,
+	MAXIMUM    = 9999, -- aribitrary limit
+	DEFAULT    = 1000000,
+}
+
+---`maxSlope` is multiplied by 1.5 at load, so 60 degrees is its actual "maximum",
+-- so has default value 15 * 1.5 = 22.5 for hovers and 90 for bots/vehicles/ships.
+local SLOPE = {
+	NONE     = 0,
+	MINIMUM  = 18,
+	MILD     = 22,
+	MODERATE = 36,
+	STEEP    = 50,
+	MAXIMUM  = 60, -- anything above 60 => 90 degrees
+	EXTREME  = 80, -- so these placeholders do nothing
+}
```

Comment:
These would be nice in a separate file so they could be `VFS.Include`d from elsewhere, for example the unit stats widget could tell if a unit is e.g. a "moderate climber" consistently.

Also perhaps it would be good if these values were in degrees (so moderate would be 54) and then divided by 1.5 in post-processing at the bottom, since that's something of an opaque technical detail that devs might not necessarily need to know.